winlogbeat/eventlog/wineventlog.go

```diff
@@ -154,8 +162,17 @@ func (l *winEventLog) Read() ([]Record, error) {
 			incrementMetric(dropReasons, err)
 			continue
 		}
+
+		r.Offset = checkpoint.EventLogState{
+			Name:         l.channelName,
+			RecordNumber: r.RecordID,
+			Timestamp:    r.TimeCreated.SystemTime,
+		}
+		if r.Offset.Bookmark, err = l.createBookmarkFromEvent(h); err != nil {
```

Comment:
Any idea as to the impact of calling this for every event? Could you run these [benchmarks](https://github.com/elastic/beats/blob/e53f42d5e1a12e0a0cbbdfa9837b0e50a91bc097/winlogbeat/eventlog/bench_test.go#L17) and share the before and after results.

Reply:
I did run some tests at first and method was taking less than 1ms to complete. However, I also found out that every event was persisted so it was necessary to always create a bookmark anyway. Maybe there was something wrong in my configuration, a side-effect of running with -N.

The benchmark results:

Before:
```
ok      github.com/elastic/beats/winlogbeat/eventlog    88.679s
ok      github.com/elastic/beats/winlogbeat/eventlog    89.105s
ok      github.com/elastic/beats/winlogbeat/eventlog    88.853s
ok      github.com/elastic/beats/winlogbeat/eventlog    89.293s
ok      github.com/elastic/beats/winlogbeat/eventlog    89.383s
```

After:
```
ok      github.com/elastic/beats/winlogbeat/eventlog    91.626s
ok      github.com/elastic/beats/winlogbeat/eventlog    90.079s
ok      github.com/elastic/beats/winlogbeat/eventlog    89.980s
ok      github.com/elastic/beats/winlogbeat/eventlog    90.663s
ok      github.com/elastic/beats/winlogbeat/eventlog    90.641s
```

It is a 1.7% overhead.

I am fine with having the bookmark XML calculated only when needed